### PR TITLE
Space

### PR DESCRIPTION
**Pull request by TheYisusGithub**

### CODE REVIEW

File: src/main/java/clickup/ui/entities/Context.java [hunk 1]
```diff
@@ -17,4 +17,28 @@
...
+
+    /**
+     * Constructor.
+     */
+    public Context() {
+        this.space = new Space();
+    }
+
+    /**
+     * Method let the ger the space.
```
Comment (MadayAlcala) [inline src/main/java/clickup/ui/entities/Context.java]: Please change the documentation according to our Code Conventions and add an empty line.

File: src/main/java/clickup/ui/entities/Context.java [hunk 2]
```diff
@@ -17,4 +17,28 @@
...
+     * @return space variable.
+     */
+    public Space getSpace() {
+        return space;
+    }
+
+    /**
+     * Mehtod to let set a space.
```
Comment (MadayAlcala) [inline src/main/java/clickup/ui/entities/Context.java]: Please change the documentation according to our Code Conventions.

File: src/main/java/clickup/ui/entities/Space.java [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+ * Context class.
+ *
+ * @author Jesus Menacho
+ * @version 1.0
+ */
+public class Space {
+    private String title;
+
+    /**
+     * Method to let get the title.
```
Comment (MadayAlcala) [inline src/main/java/clickup/ui/entities/Space.java]: Please change the documentation according to our Code Conventions  and add an empty line.

File: src/main/java/clickup/ui/entities/Space.java [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+     * @return title type string.
+     */
+    public String getTitle() {
+        return title;
+    }
+
+    /**
+     * Method to let set the title.
```
Comment (MadayAlcala) [inline src/main/java/clickup/ui/entities/Space.java]: Please change the documentation according to our Code Conventions and add an empty line.

File: src/main/java/clickup/ui/pages/SpacePage.java [hunk 5]
```diff
@@ -50,7 +58,16 @@ public void addNewSpace(String nameSpace) {
      * @param nameSpace parameter string.
      * @return boolean result.
      */
-    public boolean isFoundNameSpace(String nameSpace) {
+    public boolean isFoundNameSpace(final String nameSpace) {
         return getDriver().getPageSource().contains(nameSpace);
     }
+
+    /**
+     * Method let log out of the page.
```
Comment (MadayAlcala) [inline src/main/java/clickup/ui/pages/SpacePage.java]: Please change the documentation according to our Code Conventions.

File: src/test/java/steps/SpaceStep.java [hunk 6]
```diff
@@ -20,8 +21,17 @@
...
+    /**
+     * Constructor class.
+     * @param context type Context class.
+     */
+    public SpaceStep(Context context) {
+        this.context = context;
+    }
 
     /**
      * This method let create in the web page new name space.
```
Comment (MadayAlcala) [inline src/test/java/steps/SpaceStep.java]: Please change the documentation according to our Code Conventions.

File: src/test/java/steps/SpaceStep.java [hunk 7]
```diff
@@ -20,8 +21,17 @@
...
  * @version 1.0
  */
-public class StarStep {
-    private StarPage starPage;
+public class SpaceStep {
+    private SpacePage starPage;
+    private Context context;
+
+    /**
+     * Constructor class.
```
Comment (MadayAlcala) [inline src/test/java/steps/SpaceStep.java]: Please add add an empty line.

File: src/test/java/steps/SpaceStep.java [hunk 8]
```diff
@@ -30,8 +40,9 @@
      */
     @When("The user create a new space with the following name (.*)")
     public void createNewSpace(String nameSpace) {
-        starPage = new StarPage();
+        starPage = new SpacePage();
         starPage.addNewSpace(nameSpace);
```
Comment (MadayAlcala) [inline src/test/java/steps/SpaceStep.java]: Please change the order between line 44 and line 45, context should be first.

File: src/test/java/hook/SpaceHook.java [hunk 9]
```diff
@@ -0,0 +1,42 @@
...
+    /**
+     * Constructor initializes the class setting the context.
+     * @param context - set the class context.
+     */
+    public SpaceHook(Context context){
+        this.context=context;
+    }
+
+    /**
+     * Method to let logout.
```
Comment (MadayAlcala) [inline src/test/java/hook/SpaceHook.java]: Please change the comment according to our Code Conventions.

File: src/test/java/hook/SpaceHook.java [hunk 10]
```diff
@@ -0,0 +1,42 @@
...
+ * Hook class.
+ *
+ * @author Jesus Menacho
+ * @version 1.0
+ */
+public class SpaceHook {
+    private Context context;
+
+    /**
+     * Constructor initializes the class setting the context.
```
Comment (MadayAlcala) [inline src/test/java/hook/SpaceHook.java]: Please change the comment according to our Code Conventions and add an empty line.

File: src/main/java/clickup/ui/pages/SpacePage.java [hunk 11]
```diff
@@ -50,7 +58,16 @@ public void addNewSpace(String nameSpace) {
...
+     */
+    public void logOut() {
+        getDriver().findElement(By.xpath("//body")).sendKeys(Keys.ESCAPE);
```
Comment (MadayAlcala) [inline src/main/java/clickup/ui/pages/SpacePage.java]: Please use Page Factory for this element: "//body"

**Review by MadayAlcala — COMMENTED** [review]
Please see my comments.

File: src/main/java/clickup/ui/entities/Context.java [hunk 12]
```diff
@@ -17,4 +17,28 @@
  * @version 1.0
  */
 public class Context {
+   private Space space;
+
+    /**
+     * Constructor.
```
Comment (alszla) [inline src/main/java/clickup/ui/entities/Context.java]: Please place a more descriptive documentation.

File: src/main/java/clickup/ui/entities/Space.java [hunk 13]
```diff
@@ -0,0 +1,37 @@
...
+ * ("Confidential Information"). You shall not
+ * disclose such Confidential Information and shall use it only in
+ * accordance with the terms of the license agreement you entered into
+ * with Jalasoft.
+ */
+
+package clickup.ui.entities;
+
+/**
+ * Context class.
```
Comment (alszla) [inline src/main/java/clickup/ui/entities/Space.java]: Please change the documentation according to our Code Conventions and add an empty line.

File: src/main/java/clickup/ui/entities/Context.java (hunk 12 above)
Comment (TheYisusGithub) [reply src/main/java/clickup/ui/entities/Context.java]: Done.

File: src/main/java/clickup/ui/entities/Context.java (hunk 1 above)
Comment (TheYisusGithub) [reply src/main/java/clickup/ui/entities/Context.java]: Done.

File: src/main/java/clickup/ui/entities/Context.java (hunk 2 above)
Comment (TheYisusGithub) [reply src/main/java/clickup/ui/entities/Context.java]: Done.

File: src/main/java/clickup/ui/entities/Space.java (hunk 13 above)
Comment (TheYisusGithub) [reply src/main/java/clickup/ui/entities/Space.java]: Done.

File: src/main/java/clickup/ui/entities/Space.java (hunk 3 above)
Comment (TheYisusGithub) [reply src/main/java/clickup/ui/entities/Space.java]: Done.

File: src/main/java/clickup/ui/entities/Space.java (hunk 4 above)
Comment (TheYisusGithub) [reply src/main/java/clickup/ui/entities/Space.java]: Done.

File: src/main/java/clickup/ui/pages/SpacePage.java (hunk 5 above)
Comment (TheYisusGithub) [reply src/main/java/clickup/ui/pages/SpacePage.java]: Done.

File: src/main/java/clickup/ui/pages/SpacePage.java (hunk 11 above)
Comment (TheYisusGithub) [reply src/main/java/clickup/ui/pages/SpacePage.java]: Done.

File: src/test/java/hook/SpaceHook.java (hunk 10 above)
Comment (TheYisusGithub) [reply src/test/java/hook/SpaceHook.java]: Done.

File: src/test/java/hook/SpaceHook.java (hunk 9 above)
Comment (TheYisusGithub) [reply src/test/java/hook/SpaceHook.java]: Done.

File: src/test/java/steps/SpaceStep.java (hunk 7 above)
Comment (TheYisusGithub) [reply src/test/java/steps/SpaceStep.java]: Done.

File: src/test/java/steps/SpaceStep.java (hunk 6 above)
Comment (TheYisusGithub) [reply src/test/java/steps/SpaceStep.java]: Done.

File: src/test/java/steps/SpaceStep.java (hunk 8 above)
Comment (TheYisusGithub) [reply src/test/java/steps/SpaceStep.java]: Done.

Comment (TheYisusGithub) [reply src/test/java/steps/SpaceStep.java]: Done.

File: src/test/java/steps/SpaceStep.java (hunk 7 above)
Comment (TheYisusGithub) [reply src/test/java/steps/SpaceStep.java]: Done.

File: src/test/java/hook/SpaceHook.java (hunk 9 above)
Comment (TheYisusGithub) [reply src/test/java/hook/SpaceHook.java]: Done.

**Review by alszla — APPROVED** [review]

**Review by MadayAlcala — APPROVED** [review]

**Pull request merged**